pythran/pythonic/numpy/partial_sum.hpp

```diff
@@ -13,24 +12,53 @@ namespace pythonic
   namespace numpy
   {
 
-    template <class Op>
-    template <class E, class F, class A>
-    A _partial_sum<Op, 1>::operator()(E e, F &o, A acc)
+    template <class Op, class A>
+    template <class E, class F>
+    A _partial_sum<Op, 1, A>::operator()(E e, F &o, A acc)
     {
       for (auto const &value : e) {
-        Op{}(acc, value);
+        acc = Op{}(acc, (A)value);
```

Comment:
In this case, Op{} signature is : (A a, B b) -> decltype(a == b).
So with A = bool and B = int, it will return false. With the cast, A = bool and B = bool and it return True